v2/selective.go

```diff
@@ -101,19 +123,40 @@ func TraverseToFile(ctx context.Context, ls *ipld.LinkSystem, root cid.Cid, sele
 // TraverseV1 walks through the proposed dag traversal and writes a carv1 to the provided io.Writer
 func TraverseV1(ctx context.Context, ls *ipld.LinkSystem, root cid.Cid, selector ipld.Node, writer io.Writer, opts ...Option) (uint64, error) {
 	opts = append(opts, WithoutIndex())
+	conf := ApplyOptions(opts...)
 	tc := traversalCar{
-		size:     0,
+		size:     conf.V1Size,
 		ctx:      ctx,
 		root:     root,
 		selector: selector,
 		ls:       ls,
-		opts:     ApplyOptions(opts...),
+		opts:     conf,
 	}
 
-	len, _, err := tc.WriteV1(writer)
+	len, _, err := tc.WriteV1(tc.ctx, 0, writer)
 	return len, err
 }
 
+// CreateV1Reader creates an io.ReadSeeker that can be used to copy out the carv1 contents of a car.
+func CreateV1Reader(ctx context.Context, ls *ipld.LinkSystem, root cid.Cid, selector ipld.Node, opts ...Option) (io.ReadSeeker, error) {
```

Comment:
* Use `New` instead of `Create` for consistency with existing APIs?
* This is a root level function. Considering it applies to the "selective" read/write context, can I ask you to rename this to something less generic.

Reply:
still waiting to figure out what the partial traversal API in ipld-prime will look like. this may be a function on a traversalCar instead, depending on where we want shared cache state between different traversals to live.

Reply:
I have now lost context here and am not sure if my argument about partial traversal holds